test/functional/lua/snippet_spec.lua

```diff
@@ -230,14 +231,29 @@ describe('vim.snippet', function()
   end)
 
   it('updates snippet state when built-in completion menu is visible', function()
```

Comment:
Why are you modifying this test? These are _unit tests_ and each one should check a single functionality.

Reply:
because the snippet in this case also have newline

Reply:
Most of them do, but this specific test should just check that the built-in completion doesn't affect snippet navigation. Adding extra settings and checks will make investigating any future test failures harder, since the unit test is verifying multiple things.

Reply:
i had remove the newline in this case because we do have a single it for newlines so no need change here

Reply:
I still don't think there's need to modify this test :)

Reply:
Is newline important in this case? This is irrelevant if you just want to verify what happens when pum is visible. Otherwise, either add appropriate indent settings or modify newline's expect ❓

Reply:
But _real life_ snippets have new lines. This was a regression test added when fixing https://github.com/neovim/neovim/issues/25895, and I used the same example that echasnovski used.

I just find it odd to change this test when the bugfix is unrelated to it. The test is passing, and checks a specific behaviour independent from this PR.